config/config.go

```diff
@@ -1,39 +1,229 @@
+		Timeout                int32         `mapstructure:"timeout"`
+		MaxConn                int32         `mapstructure:"max-conn"`
+		ShardCronFrequency     time.Duration `mapstructure:"shardcronfrequency"`
+		MultiplexerPollTimeout time.Duration `mapstructure:"servermultiplexerpolltimeout"`
+		MaxClients             int           `mapstructure:"maxclients"`
+		MaxMemory              int64         `mapstructure:"maxmemory"`
+		EvictionPolicy         string        `mapstructure:"evictionpolicy"`
+		EvictionRatio          float64       `mapstructure:"evictionratio"`
+		KeysLimit              int           `mapstructure:"keyslimit"`
+		AOFFile                string        `mapstructure:"aoffile"`
+		PersistanceEnabled     bool          `mapstructure:"persistenceenabled"`
+	} `mapstructure:"server"`
+	Auth struct {
+		UserName string `mapstructure:"username"`
+		Password string `mapstructure:"password"`
+	} `mapstructure:"auth"`
+	Network struct {
+		IOBufferLength    int `mapstructure:"iobufferlength"`
+		IOBufferLengthMAX int `mapstructure:"iobufferlengthmax"`
+	} `mapstructure:"network"`
+}
 
-var (
-	ShardCronFrequency           = 1 * time.Second
-	ServerMultiplexerPollTimeout = 100 * time.Millisecond
-	ServerMaxClients             = 20000
-)
+// Default configurations for internal use
+var defaultConfig = Config{
+	Server: struct {
+		Addr                   string        `mapstructure:"addr"`
+		Port                   int           `mapstructure:"port"`
+		KeepAlive              int32         `mapstructure:"keepalive"`
+		Timeout                int32         `mapstructure:"timeout"`
+		MaxConn                int32         `mapstructure:"max-conn"`
+		ShardCronFrequency     time.Duration `mapstructure:"shardcronfrequency"`
+		MultiplexerPollTimeout time.Duration `mapstructure:"servermultiplexerpolltimeout"`
+		MaxClients             int           `mapstructure:"maxclients"`
+		MaxMemory              int64         `mapstructure:"maxmemory"`
+		EvictionPolicy         string        `mapstructure:"evictionpolicy"`
+		EvictionRatio          float64       `mapstructure:"evictionratio"`
+		KeysLimit              int           `mapstructure:"keyslimit"`
+		AOFFile                string        `mapstructure:"aoffile"`
+		PersistanceEnabled     bool          `mapstructure:"persistenceenabled"`
+	}{
+		Addr:                   constants.DefaultHost,
+		Port:                   constants.DefaultPort,
+		KeepAlive:              int32(300),
+		Timeout:                int32(300),
+		MaxConn:                int32(0),
+		ShardCronFrequency:     1 * time.Second,
+		MultiplexerPollTimeout: 100 * time.Millisecond,
+		MaxClients:             20000,
+		MaxMemory:              0,
+		EvictionPolicy:         "allkeys-lru",
+		EvictionRatio:          0.40,
+		KeysLimit:              10000,
+		AOFFile:                "./dice-master.aof",
+		PersistanceEnabled:     true,
+	},
+	Auth: struct {
+		UserName string "mapstructure:\"username\""
```

Comment:
Hello, why it's not ``mapstructure:"username"``?

Reply:
Because I used " instead of ` will fix it